src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/DashboardItemOrder.tsx

```diff
@@ -162,23 +155,8 @@ export const DashboardItemOrder = ({
           <DashboardItemOrderCertificate order={order} product={product} />
         </div>
       )}
-      {writable && (
-        <>
-          {product?.certificate_definition && order.contract?.signed_on && (
-            <DashboardItemOrderContract order={order} product={product} writable={writable} />
-          )}
-          <ContractFrame
-            order={order}
-            isOpen={contractFrameOpened}
-            onDone={() => {
-              // Set the contract in loading mode waiting for order re-fetch that will remove it.
-              setContractLoading(true);
-            }}
-            onClose={() => {
-              setContractFrameOpened(false);
-            }}
-          />
-        </>
+      {writable && isProductFetched && order.contract?.signed_on && (
```

Comment:
why not using `courseProductRelation` instead of `isProductFetched`?

Reply:
i'll renamed it `isCourseProductRelationFetched`

Reply:
Yes but why can't you just use `courseProductRelation` instead?

Reply:
i used to use the states given by `useResourceRoot`.
I imagine that `isFetched` can be different than `entity !== null || entity !== undefined`.
Checking the object can be enought but we've a state that give use exaclty what we want, so i use it.

Details:
render 01: 
* isFetched === false
* isLoading === true
* entity === undefined
render 02:
* isFetched === true
* isLoading === false
* entity === Object
=> user action that will refetch the data
render 03:
* isFetched === true
* isLoading === true
* entity === Object
render 04:
* isFetched === true
* isLoading === false
* entity === Object_2